### PR TITLE
move style tests to justfile, add them as separate github checks

### PR DESCRIPTION
**Pull request by Ekleog-NEAR**
cc @frol, with this change formatting errors should be reported basically instantaneously, while still allowing people to reproduce almost all of CI by simply running `just test-ci` locally :)

### CODE REVIEW

**Review by nagisa — APPROVED** [review]

File: Justfile [hunk 1]
```diff
@@ -15,7 +15,17 @@ ci_hack_nextest_profile := if env("CI_HACKS", "0") == "1" { "--profile ci" } els
 test *FLAGS: (test-ci FLAGS) test-extra
 
 # only the tests that are exactly the same as the ones in CI
-test-ci *FLAGS: (nextest "stable" FLAGS) (nextest "nightly" FLAGS) check-non-default check-udeps python-style-checks
+test-ci *FLAGS: check-rustfmt \
+                check-udeps \
```
[the remark this answers is not in the document]
Comment (Ekleog-NEAR) [reply Justfile:19]: Right, I learned that while looking at the CI runtimes, adjusted now, thanks :)

**Comment by Ekleog-NEAR** [comment]
It looks like splitting the style checks out made CI much faster than I even hoped in my wildest dreams: by making things more as the ecosystem expects them, it has repaired the previously broken rust cache. I also added independent rust caches for clippy, udeps and the RPC schema builds; and with all that CI now finishes within 10 minutes instead of the previous 25 minutes 🎉  

This gets our costs down from $8.96 a run to $3.17 a run (comparing the last run on this PR with the last run on master), so with 5000 runs a year, this gets us from  ~$45k a year, to ~$15k a year 🎉 

---

As a side-note, it looks like the only place rust cache still breaks is with rust nightly, where compilation resumes at proc-macro2. My guess would be it’s probably due to us downloading a nightly version on the fly, but it’s probably not worth looking too much into it for now.

**Comment by saketh-are** [comment]
The cost and time reduction is a huge win, this is awesome btw!

**Comment by Ekleog-NEAR** [comment]
Thank you for the heads’ up! I just opened https://github.com/near/nearcore/pull/10532 removing it, it’s basically been replaced with `just check-cargo-clippy` :)

**Pull request merged**